sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java

```diff
@@ -1511,35 +1511,33 @@ public Response<BinaryData> getImageGenerationsWithResponse(String deploymentOrM
      * @return the result of a successful image generation operation along with {@link Response}.
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Response<ImageGenerations> getImageGenerationsWithResponse(String deploymentOrModelName,
+    Response<ImageGenerations> getImageGenerationsWithResponse(String deploymentOrModelName,
         ImageGenerationOptions imageGenerationOptions, RequestOptions requestOptions) {
         Response<BinaryData> response = getImageGenerationsWithResponse(deploymentOrModelName,
             BinaryData.fromObject(imageGenerationOptions), requestOptions);
         return new SimpleResponse<>(response, response.getValue().toObject(ImageGenerations.class));
     }
 
-    /**
-     * Creates an image given a prompt.
-     *
-     * @param deploymentOrModelName Specifies either the model deployment name (when using Azure OpenAI) or model name
-     * (when using non-Azure OpenAI) to use for this request.
-     * @param imageGenerationOptions Represents the request data used to generate images.
-     * @throws IllegalArgumentException thrown if parameters fail the validation.
-     * @throws HttpResponseException thrown if the request is rejected by server.
-     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
-     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
-     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
-     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
-     * @return the result of a successful image generation operation.
-     */
-    @Generated
+    // @Generated
     @ServiceMethod(returns = ReturnType.SINGLE)
     public ImageGenerations getImageGenerations(String deploymentOrModelName,
         ImageGenerationOptions imageGenerationOptions) {
         // Generated convenience method for getImageGenerationsWithResponse
         RequestOptions requestOptions = new RequestOptions();
-        return getImageGenerationsWithResponse(deploymentOrModelName, BinaryData.fromObject(imageGenerationOptions),
-            requestOptions).getValue().toObject(ImageGenerations.class);
+        BinaryData imageGenerationOptionsBinaryData = BinaryData.fromObject(imageGenerationOptions);
+        // When the model name isn't passed, we are assuming dall-2 which for older versions of the service,
+        // was available through an LRO
+        if (deploymentOrModelName == null || deploymentOrModelName.isEmpty() || deploymentOrModelName.isBlank()) {
```

Comment:
NIT: [CoreUtils.isNullOrEmpty](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/CoreUtils.java#L130)

null and empty check can be replaced by a helper method from Azure Core